basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -0,0 +1,880 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <basic_autonomy/log/log.h>
+#include <basic_autonomy/helper_functions.h>
+
+namespace basic_autonomy
+{
+    namespace waypoint_generation
+    {
+         std::vector<PointSpeedPair> create_geometry_profile(const std::vector<cav_msgs::Maneuver> &maneuvers, double max_starting_downtrack,const carma_wm::WorldModelConstPtr &wm,
+                                                                   cav_msgs::VehicleState &ending_state_before_buffer,const cav_msgs::VehicleState& state,
+                                                                   const GeneralTrajConfig &general_config, const DetailedTrajConfig &detailed_config){
+            std::vector<PointSpeedPair> points_and_target_speeds;
+            
+            bool first = true;
+            ROS_DEBUG_STREAM("VehDowntrack:"<<max_starting_downtrack);
+            for(const auto &maneuver : maneuvers)
+            {
+                if(std::find(lane_follow_plugins_list.begin(), lane_follow_plugins_list.end(), general_config.trajectory_type) !=lane_follow_plugins_list.end() && maneuver.type != cav_msgs::Maneuver::LANE_FOLLOWING)
```

Comment:
Hmm, also trajectory_type seems to be a bit of a confusing name here. If it needs to match a planning plugin's name, the variable name should reflect that somehow. I'm also not 100% sure about this pattern or if it's necessary, let's have a quick chat on this to think it through and so I can make sure I understand.

Also the exception below refers to "lane following tactical plugin" when in theory this error is a result of no plugins supporting the maneuver type right?

Reply:
This list is local to the basic autonomy class. Since the plugins are either following a lane follow planning approach or lane change, the list just defines which plugin will follow which one. 
This is needed because we're checking if the correct maneuver request was sent to the plugin. I think this could be done in the tactical plugin as well. In case you think that's cleaner.